mcp/server.go

```diff
@@ -255,12 +250,54 @@ func toolForErr[In, Out any](t *Tool, h ToolHandlerFor[In, Out]) (*Tool, ToolHan
 			res = &CallToolResult{}
 		}
 		res.StructuredContent = out
+		if zero != nil {
+			// Avoid typed nil, which will serialize as JSON null.
+			// Instead, use the zero value of the non-zero
+			var z Out
+			if any(out) == any(z) { // bypass comparable check: pointers are comparable
```

Comment:
Won't this panic if Out = []int, for example?

Reply:
No, zero is only non-nil if Out is a pointer type.